### PR TITLE
Porting Adaptive Teams to Xunit

### LINKED ISSUE #4173: port dialogs.adaptive.teams

### PR DESCRIPTION
**Pull request by Virtual-Josh**
Fixes #4173

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/ConditionalTests.cs [hunk 1]
```diff
@@ -4,32 +4,41 @@
...
         }
+
+        internal static void Initialize()
+        {
+            ComponentRegistration.Add(new DeclarativeComponentRegistration());
+            ComponentRegistration.Add(new DialogsComponentRegistration());
+            ComponentRegistration.Add(new AdaptiveComponentRegistration());
+            ComponentRegistration.Add(new LanguageGenerationComponentRegistration());
+            ComponentRegistration.Add(new AdaptiveTestingComponentRegistration());
+            ComponentRegistration.Add(new TeamsComponentRegistration());
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/ConditionalTests.cs:41]: these should only be run once for the entire assembly. This refactor looks like every unit test would be changing this which impacts parallelt test execution. Can you please make this only run once?

Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/ConditionalTests.cs]: updated

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj [hunk 2]
```diff
@@ -12,6 +12,11 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.1.1" />
     <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj:13]: Do we still need MSTest here?

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj [hunk 3]
```diff
@@ -12,6 +12,11 @@
...
     <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj:14]: This one as well, I think.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj (hunk 2 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj]: removed

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj (hunk 3 above)
Comment (Virtual-Josh) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams.Tests.csproj]: removed

**Pull request merged**